shared/js/event_dispatcher.js

```diff
@@ -112,7 +118,7 @@
       var once = (parameters) => {
         eventDispatcher.off.call(this, eventName, once);
 
-        handler(parameters);
+        handler && handler(parameters);
```

Comment:
reading this, not sure I should keep this change; it was there when I returned a promise as well but now it's likely useless because of ensureValidHandler